[2023.5.0][i386] ValueError: zero-size array to reduction operation maximum which has no identity

A test run in autopkgtest, with network access.


```
489s ======================================================================
489s ERROR: test_guess_polarization (pyFAI.test.test_azimuthal_integrator.TestFlatimage.test_guess_polarization)
489s ----------------------------------------------------------------------
489s Traceback (most recent call last):
489s   File "/usr/lib/python3/dist-packages/pyFAI/test/test_azimuthal_integrator.py", line 356, in test_guess_polarization
489s     self.assertLess(abs(ai.guess_polarization(img)-0.5), 0.1)
489s                         ^^^^^^^^^^^^^^^^^^^^^^^^^^
489s   File "/usr/lib/python3/dist-packages/pyFAI/azimuthalIntegrator.py", line 3512, in guess_polarization
489s     bin_idx = valid_rings[numpy.where(nbpix.max() == nbpix)[0][-1]]
489s                                       ^^^^^^^^^^^
489s   File "/usr/lib/python3/dist-packages/numpy/core/_methods.py", line 41, in _amax
489s     return umr_maximum(a, axis, None, out, keepdims, initial, where)
489s            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
489s ValueError: zero-size array to reduction operation maximum which has no identity
489s 
489s ======================================================================
489s FAIL: test_variance_2d (pyFAI.test.test_azimuthal_integrator.TestRange.test_variance_2d)
489s This test checks that the variance is actually calculated and positive
489s ----------------------------------------------------------------------
489s Traceback (most recent call last):
489s   File "/usr/lib/python3/dist-packages/pyFAI/test/test_azimuthal_integrator.py", line 659, in test_variance_2d
489s     self.assertGreater(v.max(), 0, f"max variance is strictly positive with {res.method}")
489s AssertionError: 0.0 not greater than 0 : max variance is strictly positive with IntegrationMethod(2d int, no split, CSR, cython)
489s 

```
https://ci.debian.net/data/autopkgtest/testing/i386/p/pyfai/36183529/log.gz

--- a/pyFAI/ext/splitBBox_common.pyx
+++ b/pyFAI/ext/splitBBox_common.pyx
@@ -35,7 +35,7 @@
 
 __author__ = "Jerome Kieffer"
 __contact__ = "Jerome.kieffer@esrf.fr"
-__date__ = "07/01/2022"
+__date__ = "29/08/2023"
 __status__ = "stable"
 __license__ = "MIT"
 
@@ -294,8 +294,8 @@ class SplitBBoxIntegrator:
             position_t[::1] cpos0, cpos1, dpos0, dpos1
             mask_t[::1] cmask
             bint check_mask=False, do_split=True
-            position_t c0, c1, d0, d1, min0 = 0, max0 = 0, min1 = 0, max1 = 0, inv_area = 0
-            position_t pos0_min = 0, pos1_min = 0
+            position_t c0, c1, d0=0.0, d1=0.0, min0=0.0, max0=0.0, min1=0.0, max1=0.0, inv_area=0.0
+            position_t pos0_min=0.0, pos1_min=0.0
             position_t delta0, delta1, delta_down, delta_up, delta_left, delta_right
             position_t fbin0_min, fbin0_max, fbin1_min, fbin1_max
             Py_ssize_t i = 0, j = 0, idx = 0

--- a/pyFAI/ext/splitpixel_common.pyx
+++ b/pyFAI/ext/splitpixel_common.pyx
@@ -35,7 +35,7 @@
 
 __author__ = "Jerome Kieffer"
 __contact__ = "Jerome.kieffer@esrf.fr"
-__date__ = "07/01/2022"
+__date__ = "29/08/2023"
 __status__ = "stable"
 __license__ = "MIT"
 
@@ -288,9 +288,9 @@ class FullSplitIntegrator:
             position_t[:, ::1] v8 = numpy.empty((4,2), dtype=position_d)
             mask_t[:] cmask = self.cmask
             bint check_mask = False, chiDiscAtPi = self.chiDiscAtPi
-            position_t min0 = 0, max0 = 0, min1 = 0, max1 = 0, inv_area = 0
-            position_t pos0_min = 0, pos1_min = 0, pos1_max = 0, pos0_maxin = 0, pos1_maxin = 0
-            position_t a0 = 0, a1 = 0, b0 = 0, b1 = 0, c0 = 0, c1 = 0, d0 = 0, d1 = 0
+            position_t min0 = 0.0, max0 = 0.0, min1 = 0.0, max1 = 0.0, inv_area = 0.0
+            position_t pos0_min = 0.0, pos1_min = 0.0, pos1_max = 0.0, pos0_maxin = 0.0, pos1_maxin = 0.0
+            position_t a0 = 0.0, a1 = 0.0, b0 = 0.0, b1 = 0.0, c0 = 0.0, c1 = 0.0, d0 = 0.0, d1 = 0.0
             position_t delta0, delta1
             position_t foffset0, foffset1, sum_area, area
             Py_ssize_t i = 0, j = 0, idx = 0

